internal/consensus/metrics.go

```diff
@@ -165,6 +165,11 @@ type Metrics struct {
 	// ConsensusTime the metric to track how long the consensus takes in each block
 	//metrics: Number of seconds spent on consensus
 	ConsensusTime metrics.Histogram
+
+	// CompleteProposalTime measures how long it takes between receiving a proposal and finishing
+	// processing all of its parts. Note that this means it also includes network latency from
+	// block parts gossip
+	CompleteProposalTime metrics.Gauge
```

Comment:
@yzang2019 should we be using this in the SLO instead then?

Reply:
One issue with Gauge is that we're not sampling all the metrics since it's being scraped every 30s. We're capturing proposal times every 60 blocks ish (500ms block time) 

Would it make sense to use Histogram like ConsensusTime above?

Reply:
Ditto for the metrics here: internal/state/metrics.go

Reply:
> One issue with Gauge is that we're not sampling all the metrics since it's being scraped every 30s. We're capturing proposal times every 60 blocks ish (500ms block time)
> 
> Would it make sense to use Histogram like ConsensusTime above?

ah good to know I thought it's the opposite but you're right. Updated